app/src/org/commcare/activities/components/FormLayoutHelpers.java

```diff
@@ -22,13 +23,26 @@
 public class FormLayoutHelpers {
     public static boolean determineNumberOfValidGroupLines(FormEntryActivity activity,
                                                            Rect newRootViewDimensions,
-                                                           boolean groupNativeVisibility,
-                                                           boolean groupForcedInvisible) {
+                                                           boolean hasGroupLabel,
+                                                           boolean shouldHideGroupLabel) {
         FrameLayout header = (FrameLayout)activity.findViewById(R.id.form_entry_header);
         TextView groupLabel = ((TextView)header.findViewById(R.id.form_entry_group_label));
 
-        int contentSize = newRootViewDimensions.height();
+        int numberOfGroupLinesAllowed =
+                getNumberOfGroupLinesAllowed(groupLabel, newRootViewDimensions, activity);
+
+        if (TextViewCompat.getMaxLines(groupLabel) != numberOfGroupLinesAllowed) {
+            shouldHideGroupLabel = numberOfGroupLinesAllowed == 0;
+            groupLabel.setMaxLines(numberOfGroupLinesAllowed);
+            updateGroupViewVisibility(header, groupLabel, hasGroupLabel, shouldHideGroupLabel);
```

Comment:
As far as I can tell the bug was that the last 2 arguments in `updateGroupViewVisibility(header, groupLabel, hasGroupLabel, shouldHideGroupLabel);` were flipped.

This code was hard to understand, but during my refactors I realized that the names given didn't match the argument names for `updateGroupViewVisibility`, so I tried flipping them to match and that fixed the bug reported.